src/main/java/com/powsybl/openloadflow/sa/AcSecurityAnalysis.java

```diff
@@ -123,13 +170,25 @@ private SecurityAnalysisResult runSimulations(LfNetwork network, List<Propagated
             network.setReporter(preContSimReporter);
 
             // run pre-contingency simulation
-            AcLoadFlowResult preContingencyLoadFlowResult = new AcloadFlowEngine(context)
-                    .run();
+            AcloadFlowEngine engine = new AcloadFlowEngine(context);
+            engine.getContext().getEquationSystem();
+            if (!allSwitchesToClose.isEmpty()) {
+                var connectivity = network.getConnectivity();
+                allSwitchesToClose.stream().map(Identifiable::getId).forEach(id -> {
+                    LfBranch branch = network.getBranchById(id);
+                    branch.setDisabled(true);
```

Comment:
@annetill we only re-enable switches to close, but we should do it to for sub network that could be move out of main cc by this branch disabling?